model.py

```diff
@@ -134,51 +147,98 @@ def forward(self, query, values):
 class AttentionDecoder(nn.Module):
     def __init__(self, vocab_size, **kwargs):
         super(AttentionDecoder, self).__init__()
-        self.xent = kwargs['loss_function'] == 'xent'
+        self.xent = kwargs["loss_function"] == "xent"
         # LSTM input dimension changes with input feeding.
-        lstm_input_size = kwargs['dec_embed_size']
-        if kwargs['input_feed']:
+        lstm_input_size = kwargs["dec_embed_size"]
+        if kwargs["input_feed"]:
             lstm_input_size *= 2
 
-        self.embedding = nn.Embedding(vocab_size, kwargs['dec_embed_size'])
-        self.lstm = nn.LSTM(lstm_input_size, kwargs['dec_hidden_size'],
-            num_layers=kwargs['dec_num_layers'], dropout=kwargs['dropout'])
-        self.global_attn = GlobalAttention(query_size=kwargs['dec_hidden_size'],
-            values_size=kwargs['enc_hidden_size'] * (2 ** kwargs['enc_bidirectional']),
-            out_size=kwargs['dec_embed_size'])
+        self.embedding = nn.Embedding(vocab_size, kwargs["dec_embed_size"])
+        self.lstm = nn.LSTM(
+            lstm_input_size,
+            kwargs["dec_hidden_size"],
+            num_layers=kwargs["dec_num_layers"],
+            dropout=kwargs["dropout"],
+        )
+        self.global_attn = GlobalAttention(
+            query_size=kwargs["dec_hidden_size"],
+            values_size=kwargs["enc_hidden_size"] * (2 ** kwargs["enc_bidirectional"]),
+            out_size=kwargs["dec_embed_size"],
+        )
         if self.xent:
-            self.linear1 = nn.Linear(kwargs['dec_embed_size'], vocab_size)
-        self.dropout = nn.Dropout(kwargs['dropout'])
+            self.linear1 = nn.Linear(kwargs["dec_embed_size"], vocab_size)
+        # TODO Stop dropout at eval time.
```

Comment:
That's taken care already. When you do model.eval() if you use nn.Dropout() it will stop dropping out.
That's why I asked you put that back in the decoding function.